File: esm-order-review-main (6)/esm-order-review-main/ESM_SCM8/frontend/lib/entities.ts
export const ENTITY_OPTIONS = [
  { code: "HQ", displayName: "본사", legalName: "Silicon2 Co., Ltd.", integrated: true, orderIntegrated: false, insightIntegrated: true },
  { code: "PL", displayName: "폴란드", legalName: "SKO Sp. z o.o.", integrated: true, orderIntegrated: true, insightIntegrated: true },
  { code: "UK", displayName: "영국", legalName: "STYLEKOREAN UK LIMITED", integrated: false, orderIntegrated: false, insightIntegrated: false },
  { code: "USA", displayName: "미국", legalName: "Stylekorean Inc.", integrated: true, orderIntegrated: true, insightIntegrated: true },
  { code: "ME", displayName: "중동·두바이", legalName: "STYLEKOREAN MIDDLE EAST TRADING FZE", integrated: false, orderIntegrated: false, insightIntegrated: false },
  { code: "MX", displayName: "멕시코", legalName: "STYLEKOREAN MX S. DE R.L. DE C.V.", integrated: false, orderIntegrated: false, insightIntegrated: false },
  { code: "MY", displayName: "말레이시아", legalName: "STYLEKOREAN MY SDN. BHD.", integrated: false, orderIntegrated: false, insightIntegrated: false },
  { code: "VN", displayName: "베트남", legalName: "STYLEKOREAN VIETNAM", integrated: false, orderIntegrated: false, insightIntegrated: false }
] as const;

export type EntityCode = (typeof ENTITY_OPTIONS)[number]["code"];
export type EntityOption = (typeof ENTITY_OPTIONS)[number];

export const ENTITY_BY_CODE = Object.fromEntries(
  ENTITY_OPTIONS.map((entity) => [entity.code, entity])
) as Record<EntityCode, EntityOption>;

export function isEntityCode(value: unknown): value is EntityCode {
  return typeof value === "string" && value in ENTITY_BY_CODE;
}

export function allowedEntityOptions(codes: readonly string[]): EntityOption[] {
  const allowed = new Set(codes);
  return ENTITY_OPTIONS.filter((entity) => allowed.has(entity.code));
}

File: esm-order-review-main (6)/esm-order-review-main/ESM_SCM8/backend/entities.py
"""Canonical legal-entity definitions used by authentication and data APIs.

Permission checks always use the stable short codes below.  External API
identifiers are deliberately separate from the application-facing codes.
"""

from __future__ import annotations

from dataclasses import dataclass

from fastapi import HTTPException


@dataclass(frozen=True, slots=True)
class EntityDefinition:
    code: str
    display_name: str
    legal_name: str
    integrated: bool
    external_api_code: str | None = None


ENTITY_DEFINITIONS: tuple[EntityDefinition, ...] = (
    EntityDefinition("HQ", "본사", "Silicon2 Co., Ltd.", True, "HQ"),
    EntityDefinition("PL", "폴란드", "SKO Sp. z o.o.", True, "EU"),
    EntityDefinition("UK", "영국", "STYLEKOREAN UK LIMITED", False),
    EntityDefinition("USA", "미국", "Stylekorean Inc.", True, "US"),
    EntityDefinition("ME", "중동·두바이", "STYLEKOREAN MIDDLE EAST TRADING FZE", False),
    EntityDefinition("MX", "멕시코", "STYLEKOREAN MX S. DE R.L. DE C.V.", False),
    EntityDefinition("MY", "말레이시아", "STYLEKOREAN MY SDN. BHD.", False),
    EntityDefinition("VN", "베트남", "STYLEKOREAN VIETNAM", False),
)

ENTITY_BY_CODE = {entity.code: entity for entity in ENTITY_DEFINITIONS}
ENTITY_CODES = tuple(entity.code for entity in ENTITY_DEFINITIONS)
INTEGRATED_ENTITY_CODES = frozenset(
    entity.code for entity in ENTITY_DEFINITIONS if entity.integrated
)
ORDER_INTEGRATED_ENTITY_CODES = frozenset({"PL", "USA"})
INSIGHT_INTEGRATED_ENTITY_CODES = frozenset({"HQ", "PL", "USA"})


def normalize_entity_code(value: str | None) -> str:
    """Return a validated canonical entity code or a stable 400 response."""

    code = (value or "").strip().upper()
    if not code:
        raise HTTPException(status_code=400, detail="법인 코드를 선택해 주세요.")
    if code not in ENTITY_BY_CODE:
        raise HTTPException(status_code=400, detail="유효하지 않은 법인 코드입니다.")
    return code


def public_entity_definitions(codes: tuple[str, ...]) -> list[dict[str, object]]:
    return [
        {
            "code": definition.code,
            "display_name": definition.display_name,
            "legal_name": definition.legal_name,
            "integrated": definition.integrated,
            "order_integrated": definition.code in ORDER_INTEGRATED_ENTITY_CODES,
            "insight_integrated": definition.code in INSIGHT_INTEGRATED_ENTITY_CODES,
        }
        for code in codes
        if (definition := ENTITY_BY_CODE.get(code)) is not None
    ]
